src/main/java/edu/byu/cs/controller/WebSocketController.java

```diff
@@ -25,6 +26,7 @@ public static void onMessage(WsMessageContext ctx) {
         Session session = ctx.session;
         String message = ctx.message();
         String netId;
+        ctx.enableAutomaticPings(20, TimeUnit.SECONDS);
```

Comment:
Does this work properly even when the server is paused on a breakpoint, as if in a debug session?

My first guess is that the automatic pings would also be paused if the server is paused, which defeats the purpose of this timeout.

Reply:
I did some testing, and you're right--that's something we'll have to be aware of when debugging. Fortunately it seems like the only time it affects anything is if the server hits a breakpoint partway through grading a submission, which probably won't happen often. Either way, the original intent was just if unit tests take a long time to run, and it does fix that.

Reply:
Is there a way to _also_ increase the timeout to a large value? That would be helpful for any server debugging, should that arise. Even if we, the autograder team, don't typically use this much, the students will need an increased timeout limit because they will be extensively debugging their servers.

Reply:
I couldn't find one, and [this issue](https://github.com/javalin/javalin/issues/2329) from last month seems to suggest there isn't one. So I don't think so but I'm not sure 🤷

Reply:
> But students won't need it on the Autograder, will they? They're not debugging the code as its submitting to the autograder, the debug on their local machine, right?

@webecke My bad, let me clarify. I didn't mean to suggest that the students will be debugging on the AutoGrader. I meant to make reference to future semester where they will be building their chess projects using a version of Javalin. At that point they will be debugging code and stepping through websocket endpoints, and they will need to be able to keep their websocket connections alive.

> I couldn't find one, and https://github.com/javalin/javalin/issues/2329 from last month seems to suggest there isn't one.

@ThanGerlek Thanks for referring to that Javaline issue. [I just commented on it](https://github.com/javalin/javalin/issues/2329#issuecomment-2484739808) with my research on the WS timeout issue. From what I can tell _without having actually used Javalin myself_, the feature doesn't yet exist.

I see this as a symptom of using a premature product which is not yet fully featured. We should keep a list of this and any other weird things we discover about the platform. These kinds of details obviously don't make the entire decision, but the quantity and frequency and severity of multiple of them can influence our decision to look for a different platform as well.

We should take this is a reminder that this is just a community project. Upon reviewing the [recently merged PRs](https://github.com/javalin/javalin/pulls?q=is%3Apr+is%3Aclosed+is%3Amerged+-author%3Adependabot%5Bbot%5D), I see a total of 4 substantial PRs in the last 3 months. The professors suggested that this could be a good tool, but it may not be. We should keep our eyes and options open to switching to a different platform which may be better than Javalin.